app/metal-controller-manager/cmd/agent/main.go

```diff
@@ -247,7 +248,48 @@ func mainFunc() error {
 			shutdown(err)
 		}
 
-		var eg errgroup.Group
+		uuid, err := s.SystemInformation().UUID()
+		if err != nil {
+			shutdown(err)
+		}
+
+		var (
+			eg errgroup.Group
+			wg sync.WaitGroup
+		)
+
+		heartbeatCtx, stopHeartbeat := context.WithCancel(ctx)
+
+		heartbeatInterval := (time.Duration(createResp.GetRebootTimeout()) * time.Second) / 3
+
+		ticker := time.NewTicker(heartbeatInterval)
+
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			for {
+				select {
+				case <-ticker.C:
+					callCtx, cancel := context.WithTimeout(ctx, heartbeatInterval)
+
+					if _, err := client.WipeHeartbeat(callCtx, &api.WipeHeartbeatRequest{Uuid: uuid.String()}); err != nil {
```

Comment:
I think we should send first heartbeat asap, without waiting for the first interval so that once agent boots up, it resets the timeout

(should be as easy as moving this code out of `select` to the `for` loop before `select`